pkg/function/interface.go

```diff
@@ -10,9 +10,11 @@ import (
 
 // Datum contains methods to get the payload information.
 type Datum interface {
```

Comment:
perhaps we should split this also into DatumMetadata() and return that interface on which the UDF author can call `.Id()` and `.NumDelivered()`?